src/compas_rhino/geometry/surfaces/surface.py

```diff
@@ -27,42 +33,56 @@ class RhinoSurface(Surface):
         The parameter domain in the U direction.
     domain_v: tuple[float, float]
         The parameter domain in the V direction.
+    frame: :class:`compas.geometry.Frame`
+        The frame of the surface at the parametric origin.
     is_periodic_u: bool
         True if the surface is periodic in the U direction.
     is_periodic_v: bool
         True if the surface is periodic in the V direction.
+    native_surface: :class:`Rhino.Geometry.Surface`
+        The underlying Rhino surface object.
 
     """
 
+    def __new__(cls, *args, **kwargs):
+        # needed because the Surface.__new__ now enforces creation via alternative constructors only
+        # cls is the class that is being instantiated
+        return super(Surface, cls).__new__(cls, *args, **kwargs)
+
     def __init__(self, name=None):
         super(RhinoSurface, self).__init__(name=name)
         self._rhino_surface = None
 
+    def _get_frame_from_planesurface(self):
+        u_start = self.domain_u[0]
+        v_start = self.domain_v[0]
+        success, frame = self.native_surface.FrameAt(u_start, v_start)
+        if not success:
+            raise ValueError("Failed to get frame at u={} v={}".format(u_start, v_start))
+        return plane_to_compas_frame(frame)
```

Comment:
i don't think this is correct. control points are usually in world coordinates, not in local coordinates wrt to a frame positioned at the beginning of the domain.